src/src/main/java/fi/vrk/xroad/restadapterservice/ConsumerGateway.java

```diff
@@ -770,19 +791,19 @@ protected void serializeRequest(ServiceRequest request, SOAPElement soapRequest,
                 // Create new symmetric encrypter using of defined key length
                 Encrypter symmetricEncrypter = RESTGatewayUtil.createSymmetricEncrypter(this.keyLength);
                 // Process request parameters
-                writeGetParametersToBody(request, payload);
+                writeBodyContents(request, payload);
                 // Process request body
                 if (this.requestBody != null && !this.requestBody.isEmpty()) {
-                    if (convertPost) {
-                        convertJsonToXmlAndWriteToSoap(this.requestBody, soapRequest);
-                    } else {
-                        // send HTTP POST as an attachment
+                    if (!convertPost) {
```

Comment:
Could be combined with the enclosing "if".

Reply:
Retained the separate if-statement for (in my view) better clarity; added else-block and more comments